samples/renderer/src/winmain.cpp

```diff
@@ -0,0 +1,211 @@
+
+        auto viking_room_objects = rdmn::parse::gltf::parse(
+            "external/gltf_test_data_archive/viking_room/scene.gltf",
+            "external/gltf_test_data_archive/viking_room/scene.bin");
+
+        std::string name = "wolf";
+        rcbe::core::CoreObject wolf_object {std::move(name)};
+        {
+            rcbe::visual::texture_config tex_conf;
+            tex_conf.component_order = decltype(tex_conf.component_order)::RGBA;
+
+            rdmn::render::rasterizer_texture_config raster_tex_conf;
+            raster_tex_conf.filtering_type = decltype(raster_tex_conf.filtering_type)::linear;
+            raster_tex_conf.texture_type = decltype(raster_tex_conf.texture_type)::texture_2d;
+            raster_tex_conf.wrapping_type = decltype(raster_tex_conf.wrapping_type)::repeat;
+            raster_tex_conf.image_config = tex_conf;
+
+            std::vector<rcbe::rendering::Material::ShaderArguments> shader_args;
+            shader_args.reserve(2);
+            shader_args.push_back(
+                {"datamodel/data/rendering/shaders/default_vulkan.vert.spv",
+                 rdmn::render::ShaderType::vertex,
+                 rdmn::render::ShaderState::precompiled});
+            shader_args.push_back(
+                {"datamodel/data/rendering/shaders/default_vulkan.frag.spv",
+                 rdmn::render::ShaderType::fragment,
+                 rdmn::render::ShaderState::precompiled});
+
+            auto brick_wall_tex = rcbe::visual::make_tex_ptr(
+                "external/brick_wall_texture/file/brick_wall_texture.tga", rdmn::parse::tga::parse);
+            auto awesome_face_tex = rcbe::visual::make_tex_ptr(
+                "external/awesomeface_texture/file/awesomeface_texture.tga", rdmn::parse::tga::parse);
+
+            rcbe::visual::VisualTextureSet vts {brick_wall_tex, awesome_face_tex};
+
+            std::vector<rcbe::rendering::Material::TextureArguments> texture_args;
+            texture_args.reserve(2);
+            texture_args.push_back({brick_wall_tex, raster_tex_conf});
+            texture_args.push_back({awesome_face_tex, raster_tex_conf});
+
+            rcbe::rendering::Material::MaterialConfig config {
+                std::move(shader_args),
+                std::move(texture_args)};
+            rcbe::rendering::Material material(
+                std::move(config));
+
+            rcbe::math::Transform trn {
+                rcbe::math::Matrix4x4 {
+                    1.0, 0., 0., 1.5,
+                    0., 0., 1., 0.,
+                    0., -1., 0., 0.,
+                    0., 0., 0., 1.}};
+
+            second_mesh.transform(trn);
+
+            wolf_object.addComponent<rcbe::geometry::Mesh>(std::move(second_mesh));
+            wolf_object.addComponent<rcbe::rendering::Material>(std::move(material));
+            wolf_object.addComponent<rcbe::visual::VisualTextureSet>(std::move(vts));
```

Comment:
AFAIR this is done, so `VisualTexture` lifetime is at least as long as objects, that uses a material, containing this texture. If so, leave a `TODO` stating to implement `ResourceManager`.